packages/template-retail-react-app/app/hooks/use-basket.js

```diff
@@ -0,0 +1,56 @@
+/*
+ * Copyright (c) 2023, salesforce.com, inc.
+ * All rights reserved.
+ * SPDX-License-Identifier: BSD-3-Clause
+ * For full license text, see the LICENSE file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+
+/*
+ * Copyright (c) 2022, salesforce.com, inc.
+ * All rights reserved.
+ * SPDX-License-Identifier: BSD-3-Clause
+ * For full license text, see the LICENSE file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+
+import {useCustomerBaskets, useCustomerId, useProducts} from 'commerce-sdk-react-preview'
+
+/**
+ * This hook combine some commerce-react-sdk hooks to provide more derived data for Retail App baskets
+ * @param id - basket id to get the current used basket among baskets returned, use first basket in the array if not defined
+ * @param shouldFetchProductDetail - boolean to indicate if the baskets should fetch product details based on basket items
+ */
+export const useBasket = ({id = '', shouldFetchProductDetail = false} = {}) => {
+    const customerId = useCustomerId() || ''
+    const {
+        data: basketsData,
+        isLoading: isBasketsLoading,
+        error: isBasketsError,
+        ...restOfBaskets
+    } = useCustomerBaskets({customerId})
+    // if id is not defined, by default use the first basket in the list
+    const basket =
+        basketsData?.baskets?.find((basket) => basket.basketId === id) || basketsData?.baskets?.[0]
+    const productIds = '' ?? basket?.productItems?.map(({productId}) => productId).join(',')
```

Comment:
🤔 what's this `'' ??` for? Isn't the expression on this line always equal to empty string?

Reply:
Good catch, it is supposed to be at the end. Empty string is the fallback when the Ids is undefined